pkg/rtc/uptrackmanager.go

```diff
@@ -283,6 +283,9 @@ func (u *UpTrackManager) AddPublishedTrack(track types.MediaTrack) {
 
 func (u *UpTrackManager) RemovePublishedTrack(track types.MediaTrack) {
 	track.RemoveAllSubscribers()
+	u.lock.Lock()
+	delete(u.publishedTracks, track.ID())
```

Comment:
this function wasn't used anywhere. It seems that we need to actually remove the track from publishedTracks.

Reply:
That function is used in cloud. Agree with @cnderrauber . Would be good to let the close callback handling removal. When all publishers are closed, there is a callback that fires from up track manager IIRC and that handles RTCP channel.

But, one question is would this path cause the receiver to close. In the normal case, Pion closes the underling track and it flows back. Would that happen in this case?

Reply:
The client will receive Unpublished message and remove the track by triggering another negotiation. However, there will be a timing issue here if another participant joins and subscribes to this track.

So we need to make it immediately unavailable for subscription. What is the better approach to accomplish that?

Reply:
I guess one way to handle it is to keep a pending close list. But, looking at the on close handler, I think it is okay to remove it here.